java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -0,0 +1,247 @@
+package scenarios.crewcreation;
+
+import crew.Captain;
+import system.gameplay.engine.SystemInputScannerSingleton;
+import system.gameplay.roleplay.DiceRoller;
+
+import java.util.List;
+import java.util.Scanner;
+
+public final class CharacterCreation {
+
+    private static final Scanner scanner = SystemInputScannerSingleton.getInstance();
+
+    public static Captain runCharacterCreation() {
+        System.out.println("First things first, what is your name...?");
+        String name = scanner.nextLine();
+
+        System.out.println(
+                name + ", eh? Not the name I was expecting for a star ship captain, if I'm being honest with you..."
+        );
+
+        System.out.println("Anywho, Ability Scores are what really show what someone's made of!");
+        // TODO: Add a bit of flavor text describing each ability score
+
+        System.out.println("Well then, what might your ability scores be?");
+        System.out.println("Press any key to roll the dice and determine your potential attributes...");
+        scanner.nextLine();
+
+        List<Integer> unassignedAbilityScores = DiceRoller.rollAbilityScores();
+        // TODO: Implement functionality of the rollAbilityScore method
+
+        int meanScore = (int) unassignedAbilityScores.stream().mapToInt(i -> i).average().orElse(0);
+
+        System.out.println("Looks like these are your ability scores: " + unassignedAbilityScores);
+
+        if (meanScore == 18) {
+            System.out.println("What?!?!?");
+            System.out.println("There's no way...");
+            System.out.println("I can't believe it! You've truly won the genetic lottery!");
+        } else if (meanScore >= 14) {
+            System.out.println("Very impressive! You remind me of myself back in my younger days...");
+        } else if (meanScore >= 10) {
+            System.out.println("Too good to be bad, but too bad to be good. You must have something to prove.");
+        } else if (meanScore == 9) {
+            System.out.println("I must say, you're exceptionally average.");
+            System.out.println("You're like a ruler for the rest of the universe to measure themselves against...");
+        } else if (meanScore >= 7) {
+            System.out.println("You're a bit below average, but I guess you've never let that stop you before...");
+        } else if (meanScore >= 4) {
+            System.out.println("These stats are quite low... life must be a bit of a struggle for you, huh?");
+        } else {
+            System.out.println("Wow... I don't think I've ever seen anything quite so terrible before.");
+            System.out.println("Just so you know, there's no shame in restarting the game.");
+            System.out.println("I probably would, if I were you...");
+        }
+
+        // TODO: (A) Have player assign their ability scores
+
+        Integer[] abilityScoreValues = askAbilityScores(unassignedAbilityScores);
```

Comment:
This was a very good solution given your current knowledge, however there is a better way of doing this.

By using a data structure called [Map](https://docs.oracle.com/javase/tutorial/collections/interfaces/map.html) rather than a `List`, we can make the code easier to understand, less prone to breaking and slightly more performant. 

This is a huge oversimplification, but you can think of a map as a variation of an array, in the sense that both store a value along with a key. The difference is that in an array, the keys are always sequential integers (the indexes), whereas with a map, it is up to the programmer to decide what the keys will be for each entry to map. A `Map` object can be declared with any type of `Object` as the type for the keys, with the following syntax:
```
Map<{type of key}, {type of value}>
```
Example:
```
Map<String, Integer> assignedAbilityScores = new HashMap<>();
```
Will create an empty HashMap(most common implementation of Map in Java, don't worry about the others for now, analogous to using ArrayList for Lists) where the values will be stored as Integers and the keys stored to retrieve the corresponding values will be Strings.

Just by looking at the example above, I'm sure you already have a pretty good idea of why a Map would be a better way of representing the data in the problem that we are trying to solve. Since this is a bit out of scope for a beginner, I will handle coding the map functionality on my own, but feel free to ask questions!